### PR TITLE
Finish timestepping after happy breakdown

### LINKED ISSUE #143: `phiv_timestep` performance problems for scalar equations
**Describe the bug 🐞**

I consider this a bug since the performance difference is so severe, but I could move this to a feature request if desired.

In general, functions like `phiv_timestep` and `expv_timestep` are very fast for moderately-sized matrices. However, I'm noticing orders of magnitude in degradation, both in performance and in allocations, when the matrices are size-1.

It is possible for applications to circumvent this by manually implementing the functionality for scalar equations, but it would clearly be more convenient if the performance of these methods for scalar problems could be improved.

**Expected behavior**

Size-1 equations should be as fast as, or even faster than, larger ones.

**Minimal Reproducible Example 👇**

```julia
using ExponentialUtilities
function time(N)
        A = randn(N, N)
        b = randn(N, 1)
        @time phiv_timestep(1, A, b);
end

time(1)   # Warm up
time(1)   # 3.109664 seconds (18.85 M allocations: 18.841 GiB, 7.80% gc time)
time(2)   # 0.003811 seconds (3.81 k allocations: 406.266 KiB)
time(100) # 0.002961 seconds (261 allocations: 543.359 KiB)

```

**Environment (please complete the following information):**

  - Output of `using Pkg; Pkg.status()`

```julia
[d4d017d3] ExponentialUtilities v1.25.0
```

  - Output of `versioninfo()`

```julia
Julia Version 1.8.5
Commit 17cfb8e65ea (2023-01-08 06:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 12 virtual cores
```

### PR DESCRIPTION
**Pull request by ArneBouillon**
Fixes #143 

I looked into that issue a bit more. Expokit.jl uses the same initial time step as ExponentialUtilities.jl, but does not face the dimension-1 issue since it stops timestepping upon a happy breakdown. When using the matrix dimension as the Krylov subspace dimension,  happy breakdown will occur and there is no need for taking (potentially very small) time steps.

### CODE REVIEW

**Comment by ArneBouillon** [comment]
@ChrisRackauckas As far as I'm concerned, this one is ready for review. I added some documentation on this shortcut.

I'm unsure as to why some integration tests are failing. It seems unrelated to my changes.

File: src/krylov_phiv_adaptive.jl [hunk 1]
```diff
@@ -174,6 +179,9 @@ function phiv_timestep!(U::AbstractMatrix{T}, ts::Vector{tType}, A, B::AbstractM
         end
         # Part 2: compute ϕp(tau*A)wp using Krylov, possibly with adaptation
         arnoldi!(Ks, A, @view(W[:, end]); tol = tol, m = m, opnorm = opnorm, iop = iop)
+        if Ks.wasbreakdown
```
Comment (ChrisRackauckas) [inline src/krylov_phiv_adaptive.jl:182]: It seems you're assuming `Ks.wasbreakdown` was false to start `phiv_timestep!` but that's not necessarily true if it's the same cache now reused for a new problem. I think this needs to be set to `false` at the top each time.

**Comment by ChrisRackauckas** [comment]
See the note on re-solve. Also, does `expv_timestep!` need to be handled similarly?

**Comment by ArneBouillon** [comment]
Good point, I didn't realise the caches were reused! `expv_timestep!` calls `phiv_timestep!` under the hood, so that doesn't need extra care.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Thanks!

**Pull request merged**